python3Packages.plyer: Fix FTBFS, take up maintenance

OPNA2608 opened the pull request:
Closes #418869

NGIpkgs needs it for its Libervia packaging. In the above issue, the current maintainer (CC @rski) mentioned that they have no motivation to further maintain this package, so the NGI team will be picking it up.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [ ] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

OPNA2608 committed:
python3Packages.plyer: Fix FTBFS

OPNA2608 committed:
python3Packages.plyer: Adopt by NGI team

Current maintainer said that they have no reason to maintain this anymore.

eljamm reviewed (approved):
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 437133`

Logs: https://github.com/eljamm/nixpkgs-review-gha/actions/runs/17241854767


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.plyer</li>
    <li>python312Packages.plyer.dist</li>
    <li>python313Packages.plyer</li>
    <li>python313Packages.plyer.dist</li>
  </ul>
</details>


---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.plyer</li>
    <li>python312Packages.plyer.dist</li>
    <li>python313Packages.plyer</li>
    <li>python313Packages.plyer.dist</li>
  </ul>
</details>


---
### `x86_64-darwin`
:white_check_mark: *No rebuilds*


---
### `aarch64-darwin`
:white_check_mark: *No rebuilds*

ethancedwards8 commented:
otherwise lgtm

OPNA2608 committed:
python3Packages.plyer: Maintain abit

- format = "setuptools" -> pyproject = true, add setuptools to build-system
- --replace -> --replace-fail
- Stop deleting examples dir, has seemingly no effect on the resulting output
- Drop meta-wide "with lib"
- Use writableTmpDirAsHomeHook
- Formatting

OPNA2608 commented on pkgs/development/python-modules/plyer/default.nix:
```diff
@@ -46,18 +73,21 @@ buildPythonPackage rec {
   ];
+
   preCheck = ''
     HOME=$(mktemp -d)
```
Dropped this line, added `writableTmpDirAsHomeHook` to `nativeCheckInputs` instead.
